### PR TITLE
Implement SearchV3 user input component

### PR DESCRIPTION
**Pull request by chromakode**
For mobile, hopefully if we can implement a similar `AutosizeInput`, we can share most of this code!

### CODE REVIEW

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...1b0d034b70d9 introduce visual changes on linux. <details><summary>:mag_right: 6 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
</summary>

**Comment by chromakode** [comment]
Hmm, looks like the text alignment on Linux looks different from what I was working with on OSX. Will tweak this tomorrow.

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...d8b28cc7960b introduce visual changes on linux. <details><summary>:mag_right: 6 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...178092909ebf introduce visual changes on linux. <details><summary>:mag_right: 6 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
</summary>

**Comment by chromakode** [comment]
Let's defer on the font alignment issues and get this in -- this is very tricky. The bounding box for the text differs between Mac and Linux and this affects vertical centering. I've tried tinkering with the mac-specific hhea style metrics to no avail. Let's revisit this later (and potentially put in a mac-only style as a last resort?)

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...be94571a6e05 introduce visual changes on linux. <details><summary>:mag_right: 6 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...be94571a6e05 introduce visual changes on linux. <details><summary>:mag_right: 6 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...787410a8b89a introduce visual changes on linux. <details><summary>:mag_right: 9 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
 * Added: **Autosize_Input-Styled**  
   Autosize_Input-Styled rendered
 * Added: **Autosize_Input-Placeholder** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495918224290-42401f3b9820-787410a8b89a-linux/Autosize_Input-Placeholder-787410a8b89a.png)
 * Added: **Autosize_Input-Normal** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1495918224290-42401f3b9820-787410a8b89a-linux/Autosize_Input-Normal-787410a8b89a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 42401f3b9820...787410a8b89a introduce visual changes on linux. <details><summary>:mag_right: 9 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
 * Added: **Autosize_Input-Styled**  
   Autosize_Input-Styled rendered
 * Added: **Autosize_Input-Placeholder** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496178257326-42401f3b9820-787410a8b89a-linux/Autosize_Input-Placeholder-787410a8b89a.png)
 * Added: **Autosize_Input-Normal** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496178257326-42401f3b9820-787410a8b89a-linux/Autosize_Input-Normal-787410a8b89a.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 86da46fe836d...d5f822ba8518 introduce visual changes on linux. <details><summary>:mag_right: 9 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
 * Added: **Autosize_Input-Styled**  
   Autosize_Input-Styled rendered
 * Added: **Autosize_Input-Placeholder** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496182005296-86da46fe836d-d5f822ba8518-linux/Autosize_Input-Placeholder-d5f822ba8518.png)
 * Added: **Autosize_Input-Normal** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496182005296-86da46fe836d-d5f822ba8518-linux/Autosize_Input-Normal-d5f822ba8518.png)
</summary>

File: shared/searchv3/user-input/index.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+            type="iconfont-add"
+            style={{
+              fontSize: 12,
+              color: globalColors.blue,
+              marginLeft: globalMargins.xtiny,
+              cursor: 'pointer',
+            }}
+          />}
+        <ClickableBox style={{flex: 1, cursor: 'text', height: 24}} onClick={this._focusInput}>
+          &nbsp; {/* Material-UI child warns if no children */}
```
Comment (MarcoPolo) [inline shared/searchv3/user-input/index.js]: ? Didn't we kill material-ui?

Comment (chromakode) [inline shared/searchv3/user-input/index.js]: Happened concurrently to this -- testing to see if I can remove it!

Comment (chromakode) [inline shared/searchv3/user-input/index.js]: Yep works!

**Comment by keybase-ci-visdiff** [comment]
The commits 86da46fe836d...1a61b261fc65 introduce visual changes on linux. <details><summary>:mag_right: 9 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
 * Added: **Autosize_Input-Styled**  
   Autosize_Input-Styled rendered
 * Added: **Autosize_Input-Placeholder** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496267800561-86da46fe836d-1a61b261fc65-linux/Autosize_Input-Placeholder-1a61b261fc65.png)
 * Added: **Autosize_Input-Normal** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496267800561-86da46fe836d-1a61b261fc65-linux/Autosize_Input-Normal-1a61b261fc65.png)
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits feb643788e48...e85ffa9ac24e introduce visual changes on linux. <details><summary>:mag_right: 9 new</summary>

 * Added: **Search_V_3_user_input-Users_Wrap_Add_Button**  
   Search_V_3_user_input-Users_Wrap_Add_Button rendered
 * Added: **Search_V_3_user_input-Users_Wrap**  
   Search_V_3_user_input-Users_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text_Wrap**  
   Search_V_3_user_input-Users_Text_Wrap rendered
 * Added: **Search_V_3_user_input-Users_Text**  
   Search_V_3_user_input-Users_Text rendered
 * Added: **Search_V_3_user_input-Users_Add**  
   Search_V_3_user_input-Users_Add rendered
 * Added: **Search_V_3_user_input-Empty_Placeholder**  
   Search_V_3_user_input-Empty_Placeholder rendered
 * Added: **Autosize_Input-Styled**  
   Autosize_Input-Styled rendered
 * Added: **Autosize_Input-Placeholder** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496269497587-feb643788e48-e85ffa9ac24e-linux/Autosize_Input-Placeholder-e85ffa9ac24e.png)
 * Added: **Autosize_Input-Normal** [(view)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1496269497587-feb643788e48-e85ffa9ac24e-linux/Autosize_Input-Normal-e85ffa9ac24e.png)
</summary>

**Comment by MarcoPolo** [comment]
Looks like you need an `autosize-input.native.js` stub file

**Comment by chromakode** [comment]
Good catch @MarcoPolo, thanks! Added.

**Pull request merged**